### PR TITLE
[staging]remove all the previous select while changing module pos

### PR DESCRIPTION
**Pull request by bahl24**
Pull Request for Issue #24182  .

### Summary of Changes 
Existing code only removes immediate previous select

### Testing Instructions
Clear Cache

### Expected result
1 ordering only

### Actual result
Ordering unexpectedly adds

### CODE REVIEW

**Comment by bahl24** [comment]
Not sure if `_chzn` is particular for my system only or is valid in general.

**Comment by HLeithner** [comment]
please the last PR making changes to this file https://github.com/joomla/joomla-cms/pull/23052/files

there you see that $el.chosen('destroy') has been used, plz try this.

**Comment by bahl24** [comment]
@HLeithner Tried replacing .remove() with .chosen('destroy') but it's not working

**Comment by HLeithner** [comment]
I think you have to use:
```$("#" + $id).chosen('destroy');```

or simply try to restore the old version

```js
var $el = $("#" + $id);
if ($el) {	
  $el.chosen('destroy');
  $el.chosen();
}
```

**Comment by bahl24** [comment]
> I think you have to use:
> `$("#" + $id).chosen('destroy');`

I tried with this only. Will try with $el, but reverting it might bring back the issue due to which it was removed.

**Comment by bahl24** [comment]
> ```js
> var $el = $("#" + $id);
> if ($el) {	
>   $el.chosen('destroy');
>   $el.chosen();
> }
> ```

@HLeithner I tried to undo the changes made in https://github.com/joomla/joomla-cms/pull/23052/files, but it doesn't solve the issue and might bring back #22935. Also, is there something wrong in the solution I implemented(`.remove()`)?

**Comment by HLeithner** [comment]
if this works maybe its the correct way.

**Comment by bahl24** [comment]
@infograf768 @HLeithner Kindly test this

**Comment by infograf768** [comment]
@Fedik

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 5c90ddae3c06136c2bff8484ff204a0c32a42529<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24260">issues.joomla.org/tracker/joomla-cms/24260</a>.</sub>

**Comment by infograf768** [comment]
@HLeithner 
Works fine here whether for site or admin modules.

**Comment by alikon** [comment]
shouldn't we need the uncompressed/compressed version ?

**Comment by HLeithner** [comment]
seams there was never a compressed version...

**Comment by alikon** [comment]
maybe it's time to do it :slightly_smiling_face: or matter for another pr

**Comment by HLeithner** [comment]
another Pr would be better, I would like to merge this tomorrow if we get a second test.

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 4667c8c368c33cd740bd047284072f218f3bb6a2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24260">issues.joomla.org/tracker/joomla-cms/24260</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

File: media/system/js/moduleorder.js [hunk 1]
```diff
@@ -40,6 +40,8 @@
 
 								// Remove previous <select>, it will be recreated by writeDynaList()
 								$("#" + $id).remove();
```
Comment (Fedik) [inline media/system/js/moduleorder.js]: replace `$("#" + $id).remove();` with:
```javascript
var $previous = $("#" + $id);
if ($previous.data('chosen')){
  $previous.chosen('destroy');
}
$previous.remove();
```
then will work "correctly"

Comment (HLeithner) [reply media/system/js/moduleorder.js]: @bahl24 this sounds better please change this and we need 2 tests tomorrow so I can merge this.

**Comment by bahl24** [comment]
Okay

>
>

**Comment by bahl24** [comment]
@HLeithner @Fedik Updated, kindly test

**Comment by HLeithner** [comment]
@alikon and @infograf768  can I get new test from you? I would like to merge this for todays RC

**Comment by HLeithner** [comment]
thx

**Pull request merged**